File: 3.java
class Solution {
	public int lengthOfLongestSubstring(String s) {
		int start = 0;
		int end = 0;
		int res = 0;
		int len = s.length();
		Map<Character, Integer> newMap = new HashMap<> ();

		for (end = 0; end < len; end++) {
			if (newMap.containsKey(s.charAt(end))) {
				start=Math.max(start,newMap.get(s.charAt(end))+1);
			}
			newMap.put(s.charAt(end),end);
            res=Math.max(res, end-start+1);
            //System.out.print("start: "+start+"end "+end);
            //System.out.println("res "+res);
		}
		return res;
	}
}

File: 3.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        res=0
        start=0
        end=0
        length=len(s)
        dic={}
        for end in range(0,length):
            if s[end] in dic:
                start=max(start, dic[s[end]]+1)
            dic[s[end]]=end
            res=max(res, end-start+1)
        return res